site/docs/relations/logical_relations.md

```diff
@@ -231,6 +231,8 @@ The join operation will combine two separate inputs into a single output, based
 | Semi | Returns records from the left input. These are returned only if the records have a join partner on the right side. |
 | Anti  | Return records from the left input. These are returned only if the records do not have a join partner on the right side. |
 | Single | Returns one join partner per entry on the left input. If more than one join partner exists, there are two valid semantics. 1) Only the first match is returned. 2) The system throws an error. If there is no match between the left and right inputs, NULL is returned. |
+| Right Semi | Returns records from the right input. These are returned only if the records have a join partner on the left side. |
```

Comment:
The description (above) for single join seems incorrect according to https://15721.courses.cs.cmu.edu/spring2018/papers/16-optimizer2/hyperjoins-btw2017.pdf

Also, the description for single join is inconsistent with the other join descriptions.  From the paper we have:

> To express this in relational algebra we introduce a new operator, the single join. It behaves
nearly identical to an outer join, but reports an error if more than one join partner is found

I am going to choose to interpret "outer join" here as "left outer join" based on surrounding context.

I recommend:

```
Return all records from the left input.  For each cross input match, return a record including the data from both sides.  For any remaining non-matching records from the left input, return the left record along with nulls for the right input.  If a record from the left input has more than one match in the right input then an error must be raised.
```

Do we have any reference that the "only the first match is returned" semantic is needed by DuckDb?  I agree it is a thing that could be done but I'd rather not introduce options / variations unless there is a concrete use for them.

Reply:
@pdet Does DuckDB return only the first match?  If so we might need to make the behavior an option.

Reply:
In DuckDB the `SINGLE join` is like the `LEFT OUTER JOIN`. However it returns at most one join partner per entry on the LEFT side, and NULL if no partner is found.

There is no guarantee it will be the first one it encounters, though.